pkg/machinery/client/config/path.go

```diff
@@ -27,6 +27,17 @@ func GetDefaultPath() (string, error) {
 		return path, nil
 	}
 
+	talosSAPath := filepath.Join(constants.ServiceAccountMountPath, constants.ServiceAccountTalosconfigFilename)
+
+	_, err := os.Stat(talosSAPath)
+	if err != nil && !os.IsNotExist(err) {
```

Comment:
we should ignore more errors here, e.g. it could be 'pemission denied' if the `/var` is not readable for the user.

I would argue we should ignore all errors? but at the very least permission denied should be in the list

if there's some other error reading `/var` we can stop here and never read the `~/.talos/config`

Reply:
I also added checking for permission errors.